cli/internal/run/dry_run.go

```diff
@@ -107,38 +60,27 @@ func DryRun(
 
 	// Render the dry run as json
 	if dryRunJSON {
-		rendered, err := renderDryRunFullJSON(summary, singlePackage)
+		rendered, err := summary.FormatJSON(singlePackage)
 		if err != nil {
 			return err
 		}
 		base.UI.Output(rendered)
 		return nil
 	}
 
-	// Render the dry run as text
-	if err := displayDryTextRun(base.UI, summary, g.WorkspaceInfos, singlePackage); err != nil {
-		return err
-	}
-
-	return nil
+	return summary.FormatAndPrintText(base.UI, g.WorkspaceInfos, singlePackage)
 }
 
-func executeDryRun(ctx gocontext.Context, engine *core.Engine, g *graph.CompleteGraph, taskHashTracker *taskhash.Tracker, rs *runSpec, base *cmdutil.CmdBase, turboCache cache.Cache) ([]taskSummary, error) {
-	taskIDs := []taskSummary{}
+func executeDryRun(ctx gocontext.Context, engine *core.Engine, g *graph.CompleteGraph, taskHashTracker *taskhash.Tracker, rs *runSpec, base *cmdutil.CmdBase, turboCache cache.Cache) ([]runsummary.TaskSummary, error) {
+	taskIDs := []runsummary.TaskSummary{}
 
 	dryRunExecFunc := func(ctx gocontext.Context, packageTask *nodes.PackageTask) error {
-		hash := packageTask.Hash
-		envVars := taskEnvVarSummary{
-			Configured: packageTask.HashedEnvVars.BySource.Explicit.ToSecretHashable(),
-			Inferred:   packageTask.HashedEnvVars.BySource.Prefixed.ToSecretHashable(),
-		}
-
 		command := missingTaskLabel
 		if packageTask.Command != "" {
 			command = packageTask.Command
 		}
 
-		framework := missingFrameworkLabel
+		framework := runsummary.MissingFrameworkLabel
```

Comment:
note that this somewhat conflicts with https://github.com/vercel/turbo/pull/4050